client/src/components/History/HistoryScrollList.vue

```diff
@@ -0,0 +1,479 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faListAlt } from "@fortawesome/free-regular-svg-icons";
+import { faArrowDown, faCheckSquare, faColumns, faSignInAlt, faSquare } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { useInfiniteScroll } from "@vueuse/core";
+import { BBadge, BButton, BButtonGroup, BListGroup, BListGroupItem } from "bootstrap-vue";
+import { storeToRefs } from "pinia";
+import { computed, onMounted, onUnmounted, type PropType, type Ref, ref, watch } from "vue";
+import { useRouter } from "vue-router/composables";
+
+import type { HistoryDetailed, HistorySummary } from "@/api";
+import { useAnimationFrameResizeObserver } from "@/composables/sensors/animationFrameResizeObserver";
+import { useAnimationFrameScroll } from "@/composables/sensors/animationFrameScroll";
+import { useHistoryStore } from "@/stores/historyStore";
+import { useUserStore } from "@/stores/userStore";
+import localize from "@/utils/localization";
+
+import { HistoriesFilters } from "./HistoriesFilters";
+
+import TextSummary from "../Common/TextSummary.vue";
+import Heading from "@/components/Common/Heading.vue";
+import StatelessTags from "@/components/TagsMultiselect/StatelessTags.vue";
+import ScrollToTopButton from "@/components/ToolsList/ScrollToTopButton.vue";
+import UtcDate from "@/components/UtcDate.vue";
+
+type AdditionalOptions = "set-current" | "multi" | "center";
+type PinnedHistory = { id: string };
+
+const props = defineProps({
+    multiple: { type: Boolean, default: false },
+    selectedHistories: { type: Array as PropType<PinnedHistory[]>, default: () => [] },
+    additionalOptions: { type: Array as PropType<AdditionalOptions[]>, default: () => [] },
+    showModal: { type: Boolean, default: false },
+    inModal: { type: Boolean, default: false },
+    filter: { type: String, default: "" },
+    loading: { type: Boolean, default: false },
+});
```

Comment:
It is more straightforward to use the generic types:
https://vuejs.org/guide/typescript/composition-api#typing-component-props